assets/downloads/tutorials/extended-game-java/GameScreen.java

```diff
@@ -1,148 +1,162 @@
-	long lastDropTime;
+	Music music;
+	Sprite bucketSprite;
+	Vector2 touchPos;
+	Array<Sprite> dropSprites;
+	float dropTimer;
+	Rectangle bucketRectangle;
+	Rectangle dropRectangle;
 	int dropsGathered;
 
-	public GameScreen(final Drop gam) {
-		this.game = gam;
+	public GameScreen(final Drop game) {
+		this.game = game;
 
-		// load the images for the droplet and the bucket, 64x64 pixels each
-		dropImage = new Texture(Gdx.files.internal("droplet.png"));
-		bucketImage = new Texture(Gdx.files.internal("bucket.png"));
+		// load the images for the background, bucket and droplet
+		backgroundTexture = new Texture("background.png");
+		bucketTexture = new Texture("bucket.png");
+		dropTexture = new Texture("drop.png");
 
-		// load the drop sound effect and the rain background "music"
-		dropSound = Gdx.audio.newSound(Gdx.files.internal("drop.wav"));
-		rainMusic = Gdx.audio.newMusic(Gdx.files.internal("rain.mp3"));
-		rainMusic.setLooping(true);
+		// load the drop sound effect and background music
+		dropSound = Gdx.audio.newSound(Gdx.files.internal("drop.mp3"));
+		music = Gdx.audio.newMusic(Gdx.files.internal("music.mp3"));
+		music.setLooping(true);
+		music.setVolume(0.5F);
 
-		// create the camera and the SpriteBatch
-		camera = new OrthographicCamera();
-		camera.setToOrtho(false, 800, 480);
+		bucketSprite = new Sprite(bucketTexture);
+		bucketSprite.setSize(1, 1);
 
-		// create a Rectangle to logically represent the bucket
-		bucket = new Rectangle();
-		bucket.x = 800 / 2 - 64 / 2; // center the bucket horizontally
-		bucket.y = 20; // bottom left corner of the bucket is 20 pixels above
-						// the bottom screen edge
-		bucket.width = 64;
-		bucket.height = 64;
+		touchPos = new Vector2();
 
-		// create the raindrops array and spawn the first raindrop
-		raindrops = new Array<Rectangle>();
-		spawnRaindrop();
+		bucketRectangle = new Rectangle();
+		dropRectangle = new Rectangle();
 
+		dropSprites = new Array<>();
 	}
 
-	private void spawnRaindrop() {
-		Rectangle raindrop = new Rectangle();
-		raindrop.x = MathUtils.random(0, 800 - 64);
-		raindrop.y = 480;
-		raindrop.width = 64;
-		raindrop.height = 64;
-		raindrops.add(raindrop);
-		lastDropTime = TimeUtils.nanoTime();
+	@Override
+	public void show() {
+		// start the playback of the background music
+		// when the screen is shown
+		music.play();
 	}
 
 	@Override
 	public void render(float delta) {
-		// clear the screen with a dark blue color. The
-		// arguments to clear are the red, green
-		// blue and alpha component in the range [0,1]
-		// of the color to be used to clear the screen.
-		ScreenUtils.clear(0, 0, 0.2f, 1);
-
-		// tell the camera to update its matrices.
-		camera.update();
+		input();
+		logic();
+		draw();
+	}
 
-		// tell the SpriteBatch to render in the
-		// coordinate system specified by the camera.
-		game.batch.setProjectionMatrix(camera.combined);
+	private void input() {
+		float speed = 4f;
+		float delta = Gdx.graphics.getDeltaTime();
```

Comment:
Agree, but I took this code from previous tutorial (Simple Game) so I copy pasted it as it is for consistency (because this code is almost same like in previous tutorial)